### PR TITLE
fix: passes max_element when load index in hnswlib

### PR DESCRIPTION
**Pull request by AnneYang720**
This is the additional fix for PR #1424 .
We need to pass `max_elements` in method `load_index()`, otherwise it's reset to 0.

### CODE REVIEW

**Review by JoanFM — CHANGES_REQUESTED** [review]
Please use this PR to add the tests discussed in discord

File: tests/index/hnswlib/test_persist_data.py [hunk 1]
```diff
@@ -81,5 +81,7 @@ def test_persist_and_restore_nested(tmp_path):
 
 
 def test_persist_index_file(tmp_path):
-    _ = HnswDocumentIndex[SimpleDoc](work_dir=str(tmp_path))
-    _ = HnswDocumentIndex[SimpleDoc](work_dir=str(tmp_path))
+    store = HnswDocumentIndex[SimpleDoc](work_dir=str(tmp_path))
```
Comment (JoanFM) [inline tests/index/hnswlib/test_persist_data.py:84]: can u call them `index` better, to have a clear distinction between indexes and stores

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: tests/index/hnswlib/test_persist_data.py (hunk 1 above)
Comment (AnneYang720) [reply tests/index/hnswlib/test_persist_data.py:84]: > can u call them `index` better, to have a clear distinction between indexes and stores

`store` is used in all existing test files. Can I change them all in this PR? @JoanFM

Comment (JoanFM) [reply tests/index/hnswlib/test_persist_data.py]: yes

**Review by JoanFM — APPROVED** [review]

**Pull request merged**